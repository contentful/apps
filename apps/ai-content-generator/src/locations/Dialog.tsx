import useDialogParameters from '@hooks/dialog/useDialogParameters';
import featureConfig, { AIFeature } from '@configs/features/featureConfig';
import CommonGenerator from '@components/app/dialog/common-generator/CommonGenerator';
import GeneratorProvider from '@providers/generatorProvider';

type DialogInvocationParameters = {
  feature: AIFeature;
  entryId: string;
};

const Dialog = () => {
  const { feature, entryId, isLoading } = useDialogParameters();

  if (isLoading) {
    return null;
  }

  const getGenerator = (feature: AIFeature) => {
    switch (feature) {
      case AIFeature.TRANSLATE:
        return <CommonGenerator isTranslate />;

      case AIFeature.TITLE:
        return <CommonGenerator isTitle />;

      default:
        return <CommonGenerator />;
    }
  };

  return (
    <GeneratorProvider entryId={entryId} prompt={featureConfig[feature].prompt}>
      {getGenerator(feature)}
    </GeneratorProvider>
  );
};

export default Dialog;
export type { DialogInvocationParameters };
