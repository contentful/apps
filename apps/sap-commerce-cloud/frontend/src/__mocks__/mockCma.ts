const mockCma: any = {
  appSignedRequest: {
    create: () => ({}),
  },
  getSpace: () => ({
    getEnvironment: () => ({
      getEditorInterfaces: () => ({
        items: [],
      }),
      getContentTypes: () => ({
        items: [],
      }),
    }),
  }),
};

export { mockCma };
