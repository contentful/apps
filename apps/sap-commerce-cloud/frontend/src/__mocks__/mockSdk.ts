import { vi } from 'vitest';

export const makeSdkMock = () => ({
  ids: {
    app: 'some-app',
    space: '123214',
    environment: 'master',
  },
  hostnames: {
    webapp: 'app.contentful.com',
  },
  window: {
    startAutoResizer: vi.fn(),
  },
  field: {
    type: 'Array',
    removeValue: vi.fn(),
    onValueChanged: vi.fn((value) => value),
    onIsDisabledChanged: vi.fn((value) => value),
    getValue: vi.fn(),
    id: 'fieldId',
    setValue: vi.fn((value) => {
      return value;
    }),
  },
  space: {
    sys: {
      id: 'spaceId',
    },
    getContentTypes: vi.fn().mockReturnValue({
      items: [
        {
          sys: { id: 'some-id' },
          name: 'some-name',
          fields: [
            {
              id: 'some-id',
              name: 'some-name',
              type: 'some-type',
            },
          ],
        },
      ],
    }),
    getEditorInterfaces: vi.fn().mockReturnValue({
      items: [
        {
          sys: { contentType: { sys: { id: 'some-id' } } },
        },
      ],
    }),
  },
  dialogs: {
    openCurrentApp: vi.fn(() => {
      return ['sku1', 'sku2'];
    }),
  },
  parameters: {
    installation: {},
    invocation: {
      apiEndpoint: 'https://api.example.com',
      baseSites: 'site1,site2',
      fieldValue: [],
    },
  },
  cma: {},
  cmaAdapter: {},
  app: {
    onConfigure: vi.fn(),
    getParameters: vi.fn().mockReturnValueOnce({}),
    setReady: vi.fn(),
    getCurrentState: vi.fn(),
  },
});
