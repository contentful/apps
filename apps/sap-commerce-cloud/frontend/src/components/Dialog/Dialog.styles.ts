import { css } from 'emotion';
import tokens from '@contentful/f36-tokens';

export const styles = {
  textInput: css({ width: '250px' }),
  grid: css({ margin: `${tokens.spacingM} 0`, paddingLeft: tokens.spacingL }),
  table: css({ padding: tokens.spacingL }),
  tableCell: css({ width: '10%' }),
  pagination: css({ margin: tokens.spacingL }),
  nextButton: (page: number) => css({ marginLeft: page > 0 ? tokens.spacingL : '' }),
  selectProductsButton: css({ margin: tokens.spacingL }),
  selectButton: css({ justifySelf: 'flex-end' }),
};
