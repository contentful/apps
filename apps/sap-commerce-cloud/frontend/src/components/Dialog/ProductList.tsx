import { DialogExtensionSDK } from '@contentful/app-sdk';
import { CheckBoxFn, Product } from '../../interfaces';
import { Checkbox, IconButton, TableCell, TableRow } from '@contentful/f36-components';
import get from 'lodash/get';
import { formatProductUrl } from '../../utils';
import { DoneIcon } from '@contentful/f36-icons';

interface Props {
  sdk: DialogExtensionSDK;
  products: Product[];
  selectedProducts: string[];
  baseSite: string;
  checkboxFn: CheckBoxFn;
}

export function ProductList(props: Props) {
  const selectButtonClickEvent = (sku: string) => {
    const apiEndpoint = get(props.sdk.parameters.invocation, 'apiEndpoint', '') as string;
    props.sdk.close([formatProductUrl(apiEndpoint, props.baseSite, sku)]);
  };

  const isFieldTypeSymbol =
    (get(props.sdk.parameters.invocation, 'fieldType', '') as string) === 'Symbol';
  return (
    <>
      {props.products.map((product, index) => {
        const checkboxValue = props.selectedProducts.includes(props.baseSite + ':' + product.sku);
        return (
          <TableRow key={product.sku}>
            <TableCell>
              {isFieldTypeSymbol ? (
                <IconButton
                  variant="primary"
                  icon={<DoneIcon />}
                  onClick={() => selectButtonClickEvent(product.sku)}
                  aria-label="Select">
                  Select
                </IconButton>
              ) : (
                <Checkbox
                  id={product.sku}
                  defaultChecked={checkboxValue}
                  onChange={props.checkboxFn}
                />
              )}
            </TableCell>
            <TableCell>{product.sku}</TableCell>
            <TableCell>{product.name}</TableCell>
            <TableCell>
              <img src={product.image} alt="product" />
            </TableCell>
          </TableRow>
        );
      })}
    </>
  );
}
