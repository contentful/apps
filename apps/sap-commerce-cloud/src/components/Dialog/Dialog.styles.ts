import { css } from 'emotion';
import tokens from '@contentful/forma-36-tokens';

export const styles = {
  textInput: css({ width: '250px' }),
  grid: css({ marginTop: tokens.spacingM, paddingLeft: tokens.spacingL }),
  table: css({ padding: tokens.spacingL }),
  tableCell: css({ width: '10%' }),
  pagination: css({ margin: tokens.spacingL }),
  nextButton: (page: number) => css({ marginLeft: page > 0 ? tokens.spacingL : '' }),
  selectProductsButton: css({ margin: tokens.spacingL }),
};
