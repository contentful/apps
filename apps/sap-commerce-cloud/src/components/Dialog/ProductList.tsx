import React from 'react';
import { DialogExtensionSDK } from '@contentful/app-sdk';
import { CheckBoxFn, Product } from '../../interfaces';
import { Button, Checkbox, TableCell, TableRow } from '@contentful/forma-36-react-components';
import get from 'lodash/get';

interface Props {
  sdk: DialogExtensionSDK;
  products: Product[];
  selectedProducts: string[];
  baseSite: string;
  checkboxFn: CheckBoxFn;
}

export class ProductList extends React.Component<Props> {
  selectButtonClickEvent(sku: string) {
    const skuId = this.props.baseSite + ':' + sku;
    this.props.sdk.close([skuId]);
  }

  render() {
    const isFieldTypeSymbol =
      (get(this.props.sdk.parameters.invocation, 'fieldType', '') as string) === 'Symbol';

    return (
      <>
        {this.props.products.map((product, index) => {
          const checkboxValue = this.props.selectedProducts.includes(
            this.props.baseSite + ':' + product.sku
          );
          return (
            <TableRow key={product.sku}>
              <TableCell>
                {isFieldTypeSymbol ? (
                  <Button
                    buttonType="primary"
                    icon="Done"
                    onClick={() => this.selectButtonClickEvent(product.sku)}>
                    Select
                  </Button>
                ) : (
                  <Checkbox
                    labelText={''}
                    id={product.sku}
                    defaultChecked={checkboxValue}
                    onChange={this.props.checkboxFn}
                  />
                )}
              </TableCell>
              <TableCell>{product.sku}</TableCell>
              <TableCell>{product.name}</TableCell>
              <TableCell>
                <img src={product.image} />
              </TableCell>
            </TableRow>
          );
        })}
      </>
    );
  }
}
