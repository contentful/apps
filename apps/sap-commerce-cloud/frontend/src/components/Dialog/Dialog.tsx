import React from 'react';
import {
  Button,
  Grid,
  GridItem,
  IconButton,
  Option,
  Select,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  TextInput,
} from '@contentful/f36-components';
import { DialogAppSDK } from '@contentful/app-sdk';
import { ProductList } from './ProductList';
import { fetchProductList } from '../../api/fetchProductList';
import { fetchBaseSites } from '../../api/fetchBaseSites';
import { AppParameters, Error, Product, SAPParameters } from '../../interfaces';
import get from 'lodash/get';
import union from 'lodash/union';
import { formatProductUrl } from '../../utils';
import { styles } from './Dialog.styles';
import { cx } from '@emotion/css';
import { DoneIcon, SearchIcon } from '@contentful/f36-icons';

interface DialogProps {
  sdk: DialogAppSDK<AppParameters>;
}

interface State {
  baseSite: string;
  baseSites: string[];
  query: string;
  page: number;
  totalPages: number;
  products: Product[];
  selectedProducts: string[];
  errors: Error[];
}

export default class Dialog extends React.Component<DialogProps, State> {
  state: State = {
    baseSite: '',
    baseSites: [],
    query: '',
    page: 0,
    totalPages: 0,
    products: [],
    selectedProducts: [],
    errors: [],
  };

  componentDidMount() {
    this.loadBaseSites().then(() => {
      this.load();
    });
  }

  load = async () => {
    const { products, errors } = await fetchProductList(
      this.state.baseSite,
      this.state.query,
      this.state.page,
      this.props.sdk.parameters as SAPParameters,
      this.updateTotalPages
    );
    this.setState({
      baseSite: this.state.baseSite,
      baseSites: this.state.baseSites,
      query: this.state.query,
      products: products,
      selectedProducts: this.state.selectedProducts,
      errors: errors,
    });
  };

  loadBaseSites = async () => {
    const baseSites = await fetchBaseSites(this.props.sdk.parameters as SAPParameters);
    let finalBaseSites: string[] = [];
    const installationConfigBaseSites = get(this.props.sdk.parameters.invocation, 'baseSites', '');
    if (installationConfigBaseSites.length > 0) {
      for (const baseSite of baseSites) {
        if (installationConfigBaseSites.split(',').includes(baseSite)) {
          finalBaseSites.push(baseSite);
        }
      }
    } else {
      finalBaseSites = baseSites;
    }
    this.setState({
      baseSite: finalBaseSites[0],
      baseSites: finalBaseSites,
      query: this.state.query,
      page: this.state.page,
      products: [],
      selectedProducts: [],
    });
  };

  updateSearchTerm = (event: any) => {
    this.setState({
      ...this.state,
      query: event.target.value,
    });
  };

  updateBaseSite = (event: any) => {
    this.setState(
      {
        ...this.state,
        baseSite: event.target.value,
      },
      () => {
        this.load();
      }
    );
  };

  multiProductsCheckBoxClickEvent = (event: any) => {
    let existingProducts: string[] = this.state.selectedProducts;
    const skuId: string = event.target.id;

    if (event.target.checked) {
      if (!existingProducts.includes(skuId)) {
        const apiEndpoint = get(this.props.sdk.parameters.invocation, 'apiEndpoint', '');
        existingProducts.push(formatProductUrl(apiEndpoint, this.state.baseSite, skuId));
        this.setState({
          selectedProducts: existingProducts,
        });
      }
    } else {
      if (existingProducts.includes(skuId)) {
        const skuIndex = existingProducts.indexOf(skuId);
        existingProducts.splice(skuIndex, 1);
        this.setState({
          selectedProducts: existingProducts,
        });
      }
    }
  };

  updateTotalPages = (totalPages: number) => {
    this.setState({ totalPages });
  };

  selectMultipleProductsClickEvent = () => {
    const currentField = get(this.props.sdk.parameters.invocation, 'fieldValue', [] as string[]);
    const updatedField = union(currentField, this.state.selectedProducts);

    this.props.sdk.close(updatedField);
  };

  searchButtonClickEvent() {
    this.load();
  }

  nextPageButtonEvent = () => {
    this.setState({ page: this.state.page + 1 }, this.load);
  };

  prevPageButtonEvent = () => {
    this.setState({ page: this.state.page - 1 }, this.load);
  };

  render() {
    const isFieldTypeArray =
      (get(this.props.sdk.parameters.invocation, 'fieldType', '') as string) === 'Array';
    return (
      <>
        <Grid
          columns="1fr 1fr 1fr 1fr 1fr"
          rowGap="spacingM"
          columnGap="spacingM"
          className={styles.grid}>
          <GridItem>
            <TextInput
              type="text"
              placeholder={'Search Term...'}
              className={cx(styles.textInput, 'f36-margin-bottom--m')}
              value={this.state.query}
              onChange={this.updateSearchTerm}
              onKeyPress={(event) => {
                if (event.key === 'Enter') {
                  this.load();
                }
              }}
            />
          </GridItem>
          <GridItem>
            <Select onChange={this.updateBaseSite}>
              {this.state.baseSites.map((baseSite) => (
                <Option key={baseSite} value={baseSite}>
                  {baseSite}
                </Option>
              ))}
            </Select>
          </GridItem>
          <GridItem>
            <IconButton
              variant="primary"
              icon={<SearchIcon />}
              aria-label="search"
              onClick={() => this.searchButtonClickEvent()}>
              Search
            </IconButton>
          </GridItem>
          {isFieldTypeArray ? (
            <GridItem>
              <IconButton
                variant="primary"
                icon={<DoneIcon />}
                onClick={this.selectMultipleProductsClickEvent}
                aria-label="Select Products">
                Select Products
              </IconButton>
            </GridItem>
          ) : (
            <></>
          )}
        </Grid>

        <Table className={styles.table}>
          {this.state.errors?.length ? (
            <TableBody>
              <TableRow>
                {this.state.errors.map((error) => (
                  <TableCell key={error.message}>
                    {' '}
                    {error.type} : {error.message}
                  </TableCell>
                ))}
              </TableRow>
            </TableBody>
          ) : (
            <></>
          )}
          {this.state.products?.length ? (
            <>
              <TableHead>
                <TableRow>
                  <TableCell className={styles.tableCell}>Select</TableCell>
                  <TableCell>Code</TableCell>
                  <TableCell>Name</TableCell>
                  <TableCell>Image</TableCell>
                </TableRow>
              </TableHead>
              <TableBody>
                <ProductList
                  sdk={this.props.sdk}
                  products={this.state.products}
                  baseSite={this.state.baseSite}
                  selectedProducts={this.state.selectedProducts}
                  checkboxFn={this.multiProductsCheckBoxClickEvent}
                />
              </TableBody>
            </>
          ) : (
            <></>
          )}
        </Table>
        <div className={styles.pagination}>
          {this.state.page > 0 ? (
            <Button variant="primary" onClick={this.prevPageButtonEvent}>
              Previous
            </Button>
          ) : (
            <></>
          )}
          {this.state.page + 1 < this.state.totalPages ? (
            <Button
              variant="primary"
              className={styles.nextButton(this.state.page)}
              onClick={this.nextPageButtonEvent}>
              Next
            </Button>
          ) : (
            <></>
          )}
        </div>
        {isFieldTypeArray ? (
          <IconButton
            variant="primary"
            icon={<DoneIcon />}
            onClick={this.selectMultipleProductsClickEvent}
            aria-label="Select Products"
            className={styles.selectProductsButton}>
            Select Products
          </IconButton>
        ) : (
          <></>
        )}
      </>
    );
  }
}
