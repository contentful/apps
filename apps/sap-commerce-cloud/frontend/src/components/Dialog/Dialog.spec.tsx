import { mockCma } from '@__mocks__/mockCma';
import { makeSdkMock } from '@__mocks__/mockSdk';
import Dialog from '@components/Dialog/Dialog';
import { Product } from '@interfaces';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { vi } from 'vitest';

const mockSdk = makeSdkMock();
vi.mock('@contentful/react-apps-toolkit', () => ({
  useSDK: () => mockSdk,
  useCMA: () => mockCma,
}));

describe('Dialog', () => {
  beforeAll(() => {
    vi.mock('@api/fetchBaseSites', () => ({
      fetchBaseSites: vi.fn(() => Promise.resolve(['site1', 'site2'])),
    }));

    vi.mock('@api/fetchProductList', () => ({
      fetchProductList: vi.fn(({ updateTotalPages }) => {
        const products: Product[] = [
          {
            id: 'product1',
            name: 'Product 1',
            sku: 'P1',
            image: 'image1.jpg',
            productUrl: 'product1',
          },
          {
            id: 'product2',
            name: 'Product 2',
            sku: 'P2',
            image: 'image2.jpg',
            productUrl: 'product2',
          },
        ];
        updateTotalPages(2);
        return Promise.resolve({ products, errors: [] });
      }),
    }));
  });

  afterAll(() => {
    vi.clearAllMocks();
  });

  it('renders correctly and matches snapshot', async () => {
    const { container } = render(<Dialog />);
    await waitFor(() => screen.getByText('Product 1'));
    expect(container).toMatchSnapshot();
  });

  it('loads base sites and products on mount', async () => {
    render(<Dialog />);
    await waitFor(() => {
      expect(screen.getByText('Product 1')).toBeInTheDocument();
      expect(screen.getByText('Product 2')).toBeInTheDocument();
    });
  });

  it('updates products when base site is changed', async () => {
    render(<Dialog />);
    await waitFor(() => screen.getByText('Product 1'));

    fireEvent.change(screen.getByRole('combobox'), { target: { value: 'site2' } });
    await waitFor(() => {
      expect(screen.getByText('Product 1')).toBeInTheDocument();
      expect(screen.getByText('Product 2')).toBeInTheDocument();
    });
  });

  it('updates products when search term is entered', async () => {
    render(<Dialog />);
    await waitFor(() => screen.getByText('Product 1'));

    const searchInput = screen.getByPlaceholderText('Type to search products');
    fireEvent.change(searchInput, { target: { value: 'Product 1' } });
    fireEvent.keyPress(searchInput, { key: 'Enter', code: 'Enter' });

    await waitFor(() => {
      expect(screen.getByText('Product 1')).toBeInTheDocument();
    });
  });

  it('handles pagination correctly', async () => {
    render(<Dialog />);
    await waitFor(() => screen.getByText('Product 1'));

    fireEvent.click(screen.getByText('Next'));
    await waitFor(() => {
      expect(screen.getByText('Product 1')).toBeInTheDocument();
    });

    fireEvent.click(screen.getByText('Previous'));
    await waitFor(() => {
      expect(screen.getByText('Product 1')).toBeInTheDocument();
    });
  });
});
