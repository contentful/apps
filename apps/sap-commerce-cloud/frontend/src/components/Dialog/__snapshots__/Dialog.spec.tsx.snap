// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dialog > renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="css-167eeg5"
  >
    <div
      class="css-1vawd3v"
    >
      <input
        aria-label="Search products"
        class="f36-margin-bottom--m css-1w37lto"
        data-test-id="cf-ui-text-input"
        placeholder="Type to search products"
        type="text"
        value=""
      />
    </div>
    <div
      class="css-1vawd3v"
    >
      <div
        class="css-1jmxdak"
      >
        <select
          class="css-hcyaiw"
          data-test-id="cf-ui-select"
        >
          <option
            data-test-id="cf-ui-select-option"
            value="site1"
          >
            site1
          </option>
          <option
            data-test-id="cf-ui-select-option"
            value="site2"
          >
            site2
          </option>
        </select>
        <svg
          aria-hidden="true"
          class="css-14n9xrq"
          data-test-id="cf-ui-icon"
          role="img"
          viewBox="0 0 24 24"
        >
          <path
            d="M5.29289 8.29289C5.68342 7.90237 6.31658 7.90237 6.70711 8.29289L12 13.5858L17.2929 8.29289C17.6834 7.90237 18.3166 7.90237 18.7071 8.29289C19.0976 8.68342 19.0976 9.31658 18.7071 9.70711L12.7071 15.7071C12.3166 16.0976 11.6834 16.0976 11.2929 15.7071L5.29289 9.70711C4.90237 9.31658 4.90237 8.68342 5.29289 8.29289Z"
          />
        </svg>
      </div>
    </div>
  </div>
  <table
    cellpadding="0"
    cellspacing="0"
    class="css-1ka9h0v"
    data-test-id="cf-ui-table"
  >
    <div
      class="css-1n2mv2k"
    >
      <div
        class="css-1pcvc58"
        data-test-id="cf-ui-spinner"
      >
        <svg
          class="css-1fukqml"
          viewBox="0 0 60 51"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Loadingâ€¦
          </title>
          <g
            class="css-13v0zw6"
            transform="translate(10,33.528168)"
          >
            <g
              class="css-1ufd1q7"
            >
              <circle
                r="6"
                transform="translate(0,10.471832)"
              />
            </g>
          </g>
          <g
            class="css-ilfgln"
            transform="translate(30,33.528168)"
          >
            <g
              class="css-3oea63"
            >
              <circle
                r="6"
                transform="translate(0,10.471832)"
              />
            </g>
          </g>
          <g
            class="css-wb1brl"
            transform="translate(50,33.528168)"
          >
            <g
              class="css-fj27a4"
            >
              <circle
                r="6"
                transform="translate(0,10.471832)"
              />
            </g>
          </g>
        </svg>
      </div>
    </div>
    <thead
      class="css-1sojo49"
      data-test-id="cf-ui-table-head"
    >
      <tr
        class="css-1sydf7g"
        data-test-id="cf-ui-table-row"
      >
        <th
          class="css-1fccvds"
          data-test-id="cf-ui-table-cell"
        >
          Select
        </th>
        <th
          class="css-9p6xbs"
          data-test-id="cf-ui-table-cell"
        >
          Code
        </th>
        <th
          class="css-9p6xbs"
          data-test-id="cf-ui-table-cell"
        >
          Name
        </th>
        <th
          class="css-9p6xbs"
          data-test-id="cf-ui-table-cell"
        >
          Image
        </th>
      </tr>
    </thead>
    <tbody
      class="css-0"
      data-test-id="cf-ui-table-body"
    >
      <tr
        class="css-1sydf7g"
        data-test-id="cf-ui-table-row"
      >
        <td
          class="css-s8xoeu"
          data-test-id="cf-ui-table-cell"
        >
          <div
            class="css-1lxwves"
          >
            <label
              class="css-1k16bbs"
              data-test-id="cf-ui-checkbox"
              for="P1"
            >
              <input
                aria-label="Product 1"
                class="css-199qhtr"
                id="P1"
                role="checkbox"
                type="checkbox"
                value=""
              />
              <span
                class="css-ely5ts"
              >
                <svg
                  aria-hidden="true"
                  class="css-1xugees"
                  data-test-id="cf-ui-icon"
                  role="img"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20.7071 5.29289C21.0976 5.68342 21.0976 6.31658 20.7071 6.70711L9.70711 17.7071C9.51957 17.8946 9.26522 18 9 18C8.73478 18 8.48043 17.8946 8.29289 17.7071L3.29289 12.7071C2.90237 12.3166 2.90237 11.6834 3.29289 11.2929C3.68342 10.9024 4.31658 10.9024 4.70711 11.2929L9 15.5858L19.2929 5.29289C19.6834 4.90237 20.3166 4.90237 20.7071 5.29289Z"
                  />
                </svg>
              </span>
            </label>
          </div>
        </td>
        <td
          class="css-s8xoeu"
          data-test-id="cf-ui-table-cell"
        >
          P1
        </td>
        <td
          class="css-s8xoeu"
          data-test-id="cf-ui-table-cell"
        >
          Product 1
        </td>
        <td
          class="css-s8xoeu"
          data-test-id="cf-ui-table-cell"
        >
          <img
            alt="product"
            src="image1.jpg"
          />
        </td>
      </tr>
      <tr
        class="css-1sydf7g"
        data-test-id="cf-ui-table-row"
      >
        <td
          class="css-s8xoeu"
          data-test-id="cf-ui-table-cell"
        >
          <div
            class="css-1lxwves"
          >
            <label
              class="css-1k16bbs"
              data-test-id="cf-ui-checkbox"
              for="P2"
            >
              <input
                aria-label="Product 2"
                class="css-199qhtr"
                id="P2"
                role="checkbox"
                type="checkbox"
                value=""
              />
              <span
                class="css-ely5ts"
              >
                <svg
                  aria-hidden="true"
                  class="css-1xugees"
                  data-test-id="cf-ui-icon"
                  role="img"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20.7071 5.29289C21.0976 5.68342 21.0976 6.31658 20.7071 6.70711L9.70711 17.7071C9.51957 17.8946 9.26522 18 9 18C8.73478 18 8.48043 17.8946 8.29289 17.7071L3.29289 12.7071C2.90237 12.3166 2.90237 11.6834 3.29289 11.2929C3.68342 10.9024 4.31658 10.9024 4.70711 11.2929L9 15.5858L19.2929 5.29289C19.6834 4.90237 20.3166 4.90237 20.7071 5.29289Z"
                  />
                </svg>
              </span>
            </label>
          </div>
        </td>
        <td
          class="css-s8xoeu"
          data-test-id="cf-ui-table-cell"
        >
          P2
        </td>
        <td
          class="css-s8xoeu"
          data-test-id="cf-ui-table-cell"
        >
          Product 2
        </td>
        <td
          class="css-s8xoeu"
          data-test-id="cf-ui-table-cell"
        >
          <img
            alt="product"
            src="image2.jpg"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <div
    class="css-hvj582"
  >
    <button
      aria-label="Next page"
      class="css-1nbom3p"
      data-test-id="cf-ui-button"
      type="button"
    >
      <span
        class="css-40g50j"
      >
        Next
      </span>
    </button>
  </div>
</div>
`;
